src/components/PeoplePage.tsx

```diff
@@ -1,8 +1,59 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { getPreparedPeople } from '../utils/functions';
+import { FilterParams } from '../types/Filters';
 
 export const PeoplePage = () => {
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get(FilterParams.Query) || '';
+  const centuries = searchParams.getAll(FilterParams.Centuries) || [];
+  const sex = searchParams.get(FilterParams.Sex) || null;
+  const sort = searchParams.get(FilterParams.Sort) || null;
+  const order = searchParams.get(FilterParams.Order) || null;
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => {
+        setErrorMessage('Something went wrong');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const preparedPeople = getPreparedPeople(
+    people,
+    sex,
+    centuries,
+    query,
+    sort,
+    order,
+  );
+
+  const peopleWithParents = preparedPeople.map((person) => {
+    const mother = people.find(mom => mom.name === person.motherName);
+    const father = people.find(dad => dad.name === person.fatherName);
+
+    return { ...person, mother, father };
+  });
```

Comment:
Could we optimize this by relocating logic to our useEffect callback to perform these calculations just once?